control/control_test.go

```diff
@@ -395,61 +401,101 @@ func TestLoad(t *testing.T) {
+				})
+				Convey("with loaded plugin type as collector", func() {
+					So(lpe.plugin.PluginType, ShouldEqual, int(plugin.CollectorPluginType))
+				})
+			})
+		})
+
 		// Stop our controller so the plugins are unloaded and cleaned up from the system
 		c.Stop()
 		time.Sleep(100 * time.Millisecond)
 	} else {
-		fmt.Printf("SNAP_PATH not set. Cannot test %s plugin.\n", fixtures.PluginNameMock2)
+		fmt.Printf("SNAP_PATH not set. Cannot test loading plugins.\n")
 	}
 }
 
 func TestLoadWithSignedPlugins(t *testing.T) {
 	if fixtures.SnapPath != "" {
-		Convey("pluginControl.Load should successufully load a signed plugin with trust enabled", t, func() {
+		Convey("pluginControl.Load with trust enabled", t, func() {
 			c := New(getTestConfig())
 			c.pluginTrust = PluginTrustEnabled
-			c.signingManager = &mocksigningManager{signed: true}
-			lpe := newListenToPluginEvent()
-			c.eventManager.RegisterHandler("Control.PluginLoaded", lpe)
+			c.signingManager = &mocksigningManager{}
 			c.Start()
-			time.Sleep(100 * time.Millisecond)
-			_, err := load(c, fixtures.PluginPathMock2, "mock.asc")
-			Convey("so error on loading a signed plugin should be nil", func() {
-				So(err, ShouldBeNil)
+			Convey("Loading a signed plugin", func() {
+				_, err := load(c, fixtures.PluginPathMock2, "mock.asc")
+				Convey("Should not return an error", func() {
+					So(err, ShouldBeNil)
+				})
+			})
+			Convey("Loading an unsigned plugin", func() {
+				_, err := load(c, fixtures.PluginPathMock1)
+				Convey("Should return an error", func() {
+					So(err, ShouldNotBeNil)
+				})
 			})
 			// Stop our controller to clean up our plugin
 			c.Stop()
-			time.Sleep(100 * time.Millisecond)
 		})
-		Convey("pluginControl.Load should successfully load unsigned plugin when trust level set to warning", t, func() {
+		Convey("pluginControl.Load when trust level set to warning", t, func() {
 			c := New(getTestConfig())
 			c.pluginTrust = PluginTrustWarn
-			c.signingManager = &mocksigningManager{signed: false}
-			lpe := newListenToPluginEvent()
-			c.eventManager.RegisterHandler("Control.PluginLoaded", lpe)
+			c.signingManager = &mocksigningManager{}
 			c.Start()
-			time.Sleep(100 * time.Millisecond)
-			_, err := load(c, fixtures.PluginPathMock2)
-			Convey("so error on loading an unsigned plugin should be nil", func() {
-				So(err, ShouldBeNil)
+			Convey("Loading a signed plugin", func() {
+				_, err := load(c, fixtures.PluginPathMock2, "mock.asc")
+				Convey("Should not return an error", func() {
+					So(err, ShouldBeNil)
+				})
+			})
+			Convey("Loading an unsigned plugin", func() {
+				_, err := load(c, fixtures.PluginPathMock1)
+				Convey("Should not return an error", func() {
+					So(err, ShouldBeNil)
+				})
 			})
 			c.Stop()
-			time.Sleep(100 * time.Millisecond)
 		})
-		Convey("pluginControl.Load returns error with trust enabled and signing not validated", t, func() {
+		Convey("pluginControl.Load with trust disabled", t, func() {
```

Comment:
Thanks for detailed explanation, I was not reading the logic, just had some time to pick simple mistakes or errors :)